---
title: "harvest_date_gwas"
author: "Sophie Watts"
date: '2021-05-03'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(message = F, warning = F)

library(tidyverse)
library(snpStats)

#load phenotype data
pheno_data <- read_csv("../data/pheno_meta_data.csv")

#load indvs from genetic data
geno_samples <- read_delim("../data/abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001.nosex", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)
geno_samples <- geno_samples$X1

#filter phenotype
pheno_data <- pheno_data %>% filter(apple_id %in% geno_samples)

write.csv(pheno_data, file = "../outputs/geno_pheno_meta_data.csv", quote = F, row.names = F)

#make a list of the phenotypes to be used in the GWAS loop.

pheno_list <- colnames(pheno_data)

pheno_list <- pheno_list[c(10:46,48)]

write.table(pheno_list, "../outputs/pheno_list.txt", sep = "\t", row.names = F, col.names = F, quote = F)

```

#Run gwas for harvest date to test running R over the command line.

```{r prelim_gwas}

harvest_date_pheno <- pheno_data %>% select(apple_id, date_jul_17_harv) %>% drop_na()
write.table(harvest_date_pheno, "../outputs/harvest_date_pheno.txt", sep = "\t", row.names = F, col.names = F)

harvest_date_indvs <- harvest_date_pheno %>% mutate(FID = apple_id) %>% select(apple_id, FID)
write.table(harvest_date_indvs, "../outputs/harvest_date_indvs.txt", sep = "\t", row.names = F, col.names = F)

#Use plink to filter indv with phenotype data and to filter for MAF of 0.01.
/project/6003429/myles_lab/bin/plink_linux_x86_64/plink --file /project/6003429/myles_lab/abc_gwas/genotype_filtering/abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001 --keep harvest_date_indvs.txt --out abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001 --maf 0.01 --recode
#After filtering for accessions with harvest data and MAF 0.01 we have 255,664 SNPs and 811 accessions.

#make a raw file
/project/6003429/myles_lab/bin/plink_linux_x86_64/plink --file abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001 --out abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001 --recode A


#Make kinship matrix with TASSEL.
#!/bin/bash
#SBATCH --time=24:00:00
#SBATCH --mem=16G
#SBATCH --output=bedrocan_gwas.out
#SBATCH --job-name=bedrocan_gwas

module load java

/project/6003429/myles_lab/bin/tassel5/tasseladmin-tassel-5-standalone-68140c7ab0dc/run_pipeline.pl -plink -ped abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001.ped -map abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001.map -KinshipPlugin -method Centered_IBS -endPlugin -export abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001_kinship.txt -exportType SqrMatrix

#Run PCA with TASSEL.
#!/bin/bash
#SBATCH --time=24:00:00
#SBATCH --mem=64G
#SBATCH --output=bedrocan_gwas.out
#SBATCH --job-name=bedrocan_gwas

module load java

/project/6003429/myles_lab/bin/tassel5/tasseladmin-tassel-5-standalone-68140c7ab0dc/run_pipeline.pl -fork1 -plink -ped abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001.ped -map abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001.map -PrincipalComponentsPlugin -covariance true -endPlugin -export abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001 -runfork1


```
#Run a linear
```{r}

#The optimal mlmm GWAS included three SNPs as co-factors. We need to extract the genotypes of these three SNPs and put them in a linear model to see how much variance in harvest date they explain along with the PCs.

#load SNPs and p-vals from mlmm.
harv_date_mlmm_pvals <- read_csv("../outputs/harv_date_mlmm_pvals.csv")

#load in map file
map <- read_table2("../data/abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001.map", 
    col_names = FALSE)

harv_1 <- map %>% filter(X1 == 3) %>% filter(X4 == 30698039)
harv_2 <- map %>% filter(X1 == 3) %>% filter(X4 == 30969306)
harv_3 <- map %>% filter(X1 == 16) %>% filter(X4 == 8852649)

#save the snp names from the 'harv' table into text file. Use plink to filter ped and map then recode as a raw file.

/project/6003429/myles_lab/bin/plink_linux_x86_64/plink --file abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001 --extract harv_top_mlmm_snps.txt --out abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001_top_mlmm_snps --recode A

#load in raw file
top_mlmm_snps <- read_table2("../data/abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001_top_mlmm_snps.raw")
top_mlmm_snps <- top_mlmm_snps[,c(1,7:ncol(top_mlmm_snps))]

harv_date_pheno <- read_delim("../outputs/harvest_date_pheno.txt", 
    "\t", escape_double = FALSE, col_names = FALSE, 
    trim_ws = TRUE)

harv_date_pheno <- harv_date_pheno %>% rename(apple_id = 'X1', harvest_date = 'X2')

lm_vars <- left_join(top_mlmm_snps, harv_date_pheno, by = c("FID" = "apple_id"))

#rename snp names
lm_vars <- lm_vars %>% rename(nac = '3_30698039_vineland_nac_A', hit_2 = '3_30969306_samtools_C', hit_3 = '16_8852649_tassel_T')

#run a linear model to calculate the variance in harvest date is explained by the top SNPs used in the multi-locus mixed model.

#NAC SNP only
summary(lm(harvest_date ~ nac, data = lm_vars)) #Adjusted R-square 0.4754, p-value: < 2.2e-16

#all co-factor SNPS
summary(lm(harvest_date ~ nac + hit_2 + hit_3 , data = lm_vars)) #Adjusted R-square 0.5435, p-value: < 2.2e-16

#load in PCs
pcs <- read_delim("../data/abc_combined_maf001_sort_vineland_imputed_pheno_hetero90_maf001_harvest_date_maf001_1.txt", 
    "\t", escape_double = FALSE, trim_ws = TRUE, 
    skip = 2)

lm_vars_pcs <- left_join(lm_vars, pcs, by = c("FID" = "Taxa"))

#all co-factor SNPs and all 5 PCs
summary(lm(harvest_date ~ nac + hit_2 + hit_3 + PC1 + PC2 + PC3 + PC4 + PC5, data = lm_vars_pcs)) #Adjusted R-square 0.6342, p-value: < 2.2e-16

lm_vars_pcs$nac <- as.factor(lm_vars_pcs$nac)

ggplot(lm_vars_pcs, aes(x=nac, y=harvest_date)) + 
  geom_boxplot()


#look at how time to ripen is correlated with NAC
ripen <- pheno_data %>% select(apple_id, time_ripen_2017, flowering_jul_16_harv)

lm_ripen <- left_join(lm_vars, ripen, by = c("FID" = "apple_id"))
summary(lm(time_ripen_2017 ~ nac, data = lm_ripen)) #Adjusted R-square 0.4367, p-value: < 2.2e-16


#use snps stats to test the mode of inheritance. Ask Zoe.

#look at how pcs correlate with harvest date and all phenotypes. look over zoe's paper genome to phenome, don't want to be redundant.

#look at the cumulative pcs

```



