---
title: "top_snp_boxplots"
author: "Sophie Watts"
date: "04/10/2021"
output: html_document
---

```{r setup, include=FALSE}

knitr::opts_chunk$set(message = F, warning = F)

library(tidyverse)
library(patchwork)

```


```{r zoom_softening}
#load SNP p-values

std_firmness_change_pvals <- read_csv("../gwas_results/standard_pvals/percent_firmness_avg_17_std_pvals.csv")

firmness_change <- std_firmness_change_pvals %>% mutate(trimmed = str_sub(SNP, 2, -3)) %>% separate(trimmed, c("chr", "bp", "snp_caller"), sep="_") %>% select(chr, bp, snp_caller, pval)

firmness_change$chr <- as.numeric(as.character(firmness_change$chr))
firmness_change$bp <- as.numeric(as.character(firmness_change$bp))

#filter for chromosome 10 and region, I took a look at a zoomed out plot and decided to zoom in on 25MB to 30MB
firmness_change <- firmness_change %>% filter(chr == 10)
firmness_change <- firmness_change %>% filter(bp >= 25000000)
firmness_change <- firmness_change %>% filter(bp <= 30000000)

#change to -log10 p-values
firmness_change <- firmness_change %>% mutate(log_pval = -log10(pval))

top_snp <- firmness_change %>% filter(bp == 27438934)
pg1_marker <- firmness_change %>% filter(bp == 27285866)

#load gene models
gene_models <- read_delim("../data/gene_models_20170612.gff3.gz", 
    delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE)
gene_models <- gene_models %>% select(X1,X3,X4,X5,X9)
gene_models <- gene_models %>% rename(chromosome = X1, class = X3, start = X4, stop = X5, annotation = X9)
#look at gene models 50kb from top hit. 
gene_models <- gene_models %>% filter(class == "gene")
gene_models <- gene_models %>% filter(chromosome == "Chr10")
#only going to pull genes from a 50kb region around the top hit
gene_models <- gene_models %>% filter(start >= 27388934)
gene_models <- gene_models %>% filter(stop <= 27488934)
#save table
write.csv(gene_models, file = "../outputs/gene_annotations/soft_chr10_zoom_gene_annots.csv", quote = F, row.names = F, col.names = T)


#larger window
zoom_chr10_firmness_change <- ggplot(firmness_change, aes(x=bp, y=log_pval))+
  geom_point(size=2, alpha=0.6, stroke=0)+
  geom_point(data=mlmm_top, 
             aes(x=bp,y=log_pval), 
             color='red',
             size=2,
             stroke=0)+
  geom_point(data=pg1_marker, 
             aes(x=bp,y=log_pval), 
             color='purple',
             size=2,
             stroke=0)+
  theme_bw()+
  geom_hline(yintercept=6.625633, linetype="dashed")+
  theme(panel.border = element_blank(), axis.text=element_text(colour = "black", size=10), axis.text.x = element_text(angle = 90, hjust = 1), axis.title=element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 0.2), plot.title = element_text(size = 9))

zoom_chr10_firmness_change

ggsave("../figures/change_firmness_zoom.pdf", plot = zoom_chr10_firmness_change)

```


```{r zoom_phenolics}
# top snp 3806070

#load gene models
gene_models <- read_delim("../data/gene_models_20170612.gff3.gz", 
    delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE)
gene_models <- gene_models %>% select(X1,X3,X4,X5,X9)
gene_models <- gene_models %>% rename(chromosome = X1, class = X3, start = X4, stop = X5, annotation = X9)
gene_models <- gene_models %>% filter(class == "gene")

#look at genes in 50kb region on chromosome 15, 3806070
gene_models_15 <- gene_models %>% filter(chromosome == "Chr15")
#only going to pull genes from a 50kb region around the top hit
gene_models_15 <- gene_models_15 %>% filter(start >= 3756070)
gene_models_15 <- gene_models_15 %>% filter(stop <= 3856070)
#save table
write.csv(gene_models_15, file = "../outputs/gene_annotations/tpc_chr15_zoom_gene_annots.csv", quote = F, row.names = F, col.names = T)

#look at genes in region on chromosome 16, 5256210
gene_models_16 <- gene_models %>% filter(chromosome == "Chr16")
#only going to pull genes from a 50kb region around the top hit
gene_models_16 <- gene_models_16 %>% filter(start >= 5206210)
gene_models_16 <- gene_models_16 %>% filter(stop <= 5306210)
#save table
write.csv(gene_models_16, file = "../outputs/gene_annotations/tpc_chr16_std_zoom_gene_annots.csv", quote = F, row.names = F, col.names = T)

#plot zooms
zoom_chr15_tpc <- ggplot(tpc_15, aes(x=bp, y=log_pval))+
  geom_point(size=2, alpha=0.6, stroke=0)+
  geom_point(data=mlmm_top_15, 
             aes(x=bp,y=log_pval), 
             color='red',
             size=2,
             stroke=0)+
  geom_point(data=mlmm_top_15_bonus, 
             aes(x=bp,y=log_pval), 
             color='red',
             size=2,
             stroke=0)+
  theme_bw()+
  geom_hline(yintercept=6.625633, linetype="dashed")+
  theme(panel.border = element_blank(), axis.text=element_text(colour = "black", size=10), axis.text.x = element_text(angle = 90, hjust = 1), axis.title=element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 0.2), plot.title = element_text(size = 9))
ggsave("../figures/zoom_chr15_tpc.pdf", plot = zoom_chr15_tpc)

zoom_chr16_tpc <- ggplot(tpc_16, aes(x=bp, y=log_pval))+
  geom_point(size=2, alpha=0.6, stroke=0)+
  geom_point(data=mlmm_top_16, 
             aes(x=bp,y=log_pval), 
             color='red',
             size=2,
             stroke=0)+
  theme_bw()+
  geom_hline(yintercept=6.625633, linetype="dashed")+
  theme(panel.border = element_blank(), axis.text=element_text(colour = "black", size=10), axis.text.x = element_text(angle = 90, hjust = 1), axis.title=element_blank(), panel.grid.major = element_blank(), panel.grid.minor = element_blank(), axis.line = element_line(colour = "black", size = 0.2), plot.title = element_text(size = 9))
ggsave("../figures/zoom_chr16_tpc.pdf", plot = zoom_chr16_tpc)

```


```{r zoom_firm_stor}
#load SNP p-values

firmness_stor_std_pvals <- read_csv("../gwas_results/standard_pvals/firmness_avg_17_stor_std_pvals.csv")

firmness_stor_std_pvals <- firmness_stor_std_pvals %>% mutate(trimmed = str_sub(SNP, 2, -3)) %>% separate(trimmed, c("chr", "bp", "snp_caller"), sep="_") %>% select(chr, bp, snp_caller, pval)

firmness_stor_std_pvals$chr <- as.numeric(as.character(firmness_stor_std_pvals$chr))
firmness_stor_std_pvals$bp <- as.numeric(as.character(firmness_stor_std_pvals$bp))

#change to -log10 p-values
firmness_stor_std_pvals <- firmness_stor_std_pvals %>% mutate(log_pval = -log10(pval))

#filter for chromosome 16, top hit is 9235229
firm_stor_16 <- firmness_stor_std_pvals %>% filter(chr == 16)
firm_stor_16 <- firm_stor_16 %>% filter(bp >= 6735229)
firm_stor_16 <- firm_stor_16 %>% filter(bp <= 11735229)

#load gene models
gene_models <- read_delim("../data/gene_models_20170612.gff3.gz", 
    delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE)
gene_models <- gene_models %>% select(X1,X3,X4,X5,X9)
gene_models <- gene_models %>% rename(chromosome = X1, class = X3, start = X4, stop = X5, annotation = X9)
gene_models <- gene_models %>% filter(class == "gene")

#look at genes in 50kb region on chromosome 16, 9235229
gene_models_16 <- gene_models %>% filter(chromosome == "Chr16")
#only going to pull genes from a 50kb region around the top hit
gene_models_16 <- gene_models_16 %>% filter(start >= 9185229)
gene_models_16 <- gene_models_16 %>% filter(stop <= 9285229)
#save table
write.csv(gene_models_16, file = "../outputs/gene_annotations/firm_stor_chr16_zoom_gene_annots.csv", quote = F, row.names = F, col.names = T)


```


```{r zoom_nac}

#load SNP p-values

#load gene models
gene_models <- read_delim("../data/gene_models_20170612.gff3.gz", 
    delim = "\t", escape_double = FALSE, 
    col_names = FALSE, trim_ws = TRUE)
gene_models <- gene_models %>% select(X1,X3,X4,X5,X9)
gene_models <- gene_models %>% rename(chromosome = X1, class = X3, start = X4, stop = X5, annotation = X9)
gene_models <- gene_models %>% filter(class == "gene")

#look at genes in 50kb region on chromosome 3, 30,698,039
gene_models_3 <- gene_models %>% filter(chromosome == "Chr03")
#only going to pull genes from a 50kb region around the top hit
gene_models_3 <- gene_models_3 %>% filter(start >= 30648039)
gene_models_3 <- gene_models_3 %>% filter(stop <= 30748039)
#save table
write.csv(gene_models_3, file = "../outputs/gene_annotations/nac_chr3_zoom_gene_annots.csv", quote = F, row.names = F, col.names = T)

```
